src/devices/video/ef9365.cpp

```diff
@@ -965,186 +909,182 @@ void ef9365_device::screen_scanning( int force_clear )
-			break;
+				busy_cycles = bitplane_xres * bitplane_yres;
+				break;
 			case 0x5: // X and Y registers reset to 0
 				set_x_reg(0);
 				set_y_reg(0);
-				set_busy_flag( cycles_to_us( 4 ) ); // Timing to check on the real hardware
-			break;
+				break;
 			case 0x6: // X and Y registers reset to 0 and clear screen
 				set_x_reg(0);
 				set_y_reg(0);
 				screen_scanning(1);
-				set_busy_flag( cycles_to_us( bitplane_xres*bitplane_yres ) ); // Timing to check on the real hardware
-			break;
+				busy_cycles = bitplane_xres * bitplane_yres;
+				break;
 			case 0x7: // Clear screen, set CSIZE to code "minsize". All other registers reset to 0
 				m_registers[EF936X_REG_CSIZE] = 0x11;
 				screen_scanning(1);
-				set_busy_flag( cycles_to_us( bitplane_xres*bitplane_yres ) ); // Timing to check on the real hardware
-			break;
+				busy_cycles = bitplane_xres * bitplane_yres;
+				break;
 			case 0x8: // Light-pen initialization (/White forced low)
-				set_busy_flag( cycles_to_us( 4 ) ); // Timing to check on the real hardware
-			break;
+				break;
 			case 0x9: // Light-pen initialization
-				set_busy_flag( cycles_to_us( 4 ) ); // Timing to check on the real hardware
-			break;
-			case 0xA: // 5x8 block drawing (size according to CSIZE)
-				busy_cycles = draw_character( 0x00 , 1 , 0 );
-				set_busy_flag( cycles_to_us( busy_cycles ) );
-			break;
-			case 0xB: // 4x4 block drawing (size according to CSIZE)
-				busy_cycles = draw_character( 0x00 , 1 , 1 );
-				set_busy_flag( cycles_to_us( busy_cycles ) );
-			break;
-			case 0xC: // Screen scanning : pen or Eraser as defined by CTRL1
+				break;
+			case 0xa: // 5x8 block drawing (size according to CSIZE)
+				busy_cycles = draw_character(0x00, true, false);
+				break;
+			case 0xb: // 4x4 block drawing (size according to CSIZE)
+				busy_cycles = draw_character(0x00, true, true);
+				break;
+			case 0xc: // Screen scanning : pen or Eraser as defined by CTRL1
 				screen_scanning(0);
-				set_busy_flag( cycles_to_us( bitplane_xres*bitplane_yres ) ); // Timing to check on the real hardware
-			break;
-			case 0xD: // X  reset to 0
+				busy_cycles = bitplane_xres * bitplane_yres;
+				break;
+			case 0xd: // X  reset to 0
 				set_x_reg(0);
-				set_busy_flag( cycles_to_us( 4 ) ); // Timing to check on the real hardware
-			break;
-			case 0xE: // Y  reset to 0
+				break;
+			case 0xe: // Y  reset to 0
 				set_y_reg(0);
-				set_busy_flag( cycles_to_us( 4 ) ); // Timing to check on the real hardware
-			break;
-			case 0xF: // Direct image memory access request for the next free cycle.
-				set_busy_flag( cycles_to_us( 64 ) ); // Timing to check on the real hardware
+				break;
+			case 0xf: // Direct image memory access request for the next free cycle.
+				busy_cycles = 64;
 				dump_bitplanes_word();
-			break;
+				break;
 			default:
 				logerror("Unemulated EF9365 cmd: %02x\n", cmd);
+				break;
 		}
+
+		set_busy_flag(cycles_to_us(busy_cycles));
```

Comment:
This causes a change in behaviour – unemulated commands now set the busy flag for four cycles, while previously they wouldn’t.

Reply:
Added handling to set `busy_cycles` to 0 for unemulated commands and avoid calling `set_busy_flag` if it is 0.